### PR TITLE
Improve error reporting on the REST level (PLT-9050)

### PR DESCRIPTION
**Pull request by paluh**
* Add more structured information to some of the reports
* Encode every error structure explicite - drop generic helper which leaked internal implementation details.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
    - [x] Operables are updated with changes to executable command line options.
    - [ ] Deploy charts updated with changes to operables.
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - [x] Review required
        - [ ] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

**Comment by paluh** [comment]
Jamie, I'm not sure if I should just update the golden tests or not.

File: marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/BuildConstraintsSpec.hs [hunk 1]
```diff
@@ -510,20 +509,15 @@ buildApplyInputsConstraintsSpec =
...
-                  tipTime < minTime
-              else
-                if "TEIntervalError (InvalidInterval " `isPrefixOf` message
-                  then
-                    counterexample "Rounding off causes the timeout to fall at the tip if the interval was invalid (effectively empty)." $
-                      tipSlot' == toSlot timeout || marloweContract == whenWhenCloseContract timeout timeout' && tipSlot' == toSlot timeout'
-                  else counterexample "Unexpected transaction failure" False
+          Left (ApplyInputsConstraintsBuildupFailed (MarloweComputeTransactionFailed (Semantics.TEIntervalError intervalError))) ->
+            case intervalError of
+              (Semantics.IntervalInPastError _ _) -> counterexample "The tip is in the past if the interval was in the past." $ tipTime < minTime
```
Comment (jhbertra) [inline marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/BuildConstraintsSpec.hs:517]: Thank you for improving these types!

**Review by jhbertra — APPROVED** [review]
Thanks!

**Comment by jhbertra** [comment]
> Jamie, I'm not sure if I should just update the golden tests or not.

Yes, please!

**Pull request merged**